eng/pipelines/templates/stages/archetype-sdk-tests.yml

```diff
@@ -107,131 +95,74 @@ parameters:
+                    JobTemplatePath: /eng/pipelines/templates/jobs/live.tests.yml
+                    OsVmImage: azsdk-pool-mms-ubuntu-2004-1espt
+                    Pool: azsdk-pool-mms-ubuntu-2004-general
+                    AdditionalParameters:
+                      ServiceDirectory: ${{ parameters.ServiceDirectory }}
+                      TestResourceDirectories: ${{ parameters.TestResourceDirectories }}
+                      PreSteps:
+                        - ${{ parameters.PlatformPreSteps }}
+                        - ${{ parameters.PreSteps }}
+                      PostSteps:
+                        - ${{ parameters.PlatformPostSteps }}
+                        - ${{ parameters.PostSteps }}
+                      EnvVars: ${{ parameters.EnvVars }}
+                      MaxParallel: ${{ parameters.MaxParallel }}
+                      BeforeTestSteps: ${{ parameters.BeforeTestSteps }}
+                      AfterTestSteps: ${{ parameters.AfterTestSteps }}
+                      AdditionalTestArgs: ${{ parameters.AdditionalTestArgs }}
+                      BuildTargetingString: ${{ package }}
+                      TestMarkArgument: ${{ parameters.TestMarkArgument }}
+                      InjectedPackages: ${{ parameters.InjectedPackages }}
+                      BuildDocs: ${{ parameters.BuildDocs }}
+                      TestTimeoutInMinutes: ${{ parameters.TestTimeoutInMinutes }}
+                      TestProxy: ${{ parameters.TestProxy }}
+                      ToxTestEnv: ${{ parameters.ToxTestEnv }}
+                      UseFederatedAuth: ${{ parameters.UseFederatedAuth }}
+                    MatrixConfigs:
+                      # Enumerate platforms and additional platforms based on supported clouds (sparse platform<-->cloud matrix).
+                      - ${{ each config in parameters.MatrixConfigs }}:
+                          - ${{ config }}
+                      - ${{ each config in parameters.AdditionalMatrixConfigs }}:
+                          - ${{ config }}
+                    MatrixFilters:
+                      - ${{ each cloudFilter in cloud.value.MatrixFilters }}:
+                          - ${{ cloudFilter }}
+                      - ${{ parameters.MatrixFilters }}
+                    MatrixReplace:
+                      - ${{ each cloudReplace in cloud.value.MatrixReplace }}:
+                          - ${{ cloudReplace }}
+                      - ${{ parameters.MatrixReplace }}
+                    CloudConfig:
+                      SubscriptionConfiguration: ${{ cloud.value.SubscriptionConfiguration }}
+                      SubscriptionConfigurations: ${{ cloud.value.SubscriptionConfigurations }}
+                      Location: ${{ coalesce(cloud.value.Location, parameters.Location) }}
+                      Cloud: ${{ cloud.key }}
+                      ServiceConnection: ${{ coalesce(cloud.value.ServiceConnection, lower(format('azure-sdk-tests-{0}', cloud.key))) }}
```

Comment:
@benbp let me know what you think about this convention. It would allow us to define well known connections and not need to specify ServiceConnection explicitly all the time.

Reply:
Yeah I'm fine with this. Not having the options enumerated in the config could be annoying, but I like the parameter definition simplicity we get.